spinalcordtoolbox/scripts/sct_deepseg.py

```diff
@@ -530,46 +530,39 @@ def main(argv: Sequence[str]):
+                        "CodeURL": f"https://github.com/spinalcordtoolbox/spinalcordtoolbox/"
+                                f"blob/{_git_info()[1].strip('*')}/spinalcordtoolbox/scripts/sct_deepseg.py",
+                        "ModelURL": source_dict["model_urls"],
+                    }
+                ]
+            }
+            with open(splitext(fname_seg)[0] + ".json", "w") as fp:
+                json.dump(sidecar_json, fp, indent=4)
+
         # Use the result of the current model as additional input of the next model
         fname_prior = fname_seg
 
-        # write JSON sidecar file
-        source_path = os.path.join(path_model, "source.json")
-        if os.path.isfile(source_path):
-            with open(source_path, "r") as fp:
-                source_dict = json.load(fp)
-        sidecar_json = {
-            'GeneratedBy': [
-                {
-                    "Name": f"spinalcordtoolbox: sct_deepseg {' '.join(os.path.basename(arg) for arg in argv)}",
-                    "Version": __version__,
-                    "CodeURL": f"https://github.com/spinalcordtoolbox/spinalcordtoolbox/"
-                               f"blob/{_git_info()[1].strip('*')}/spinalcordtoolbox/scripts/sct_deepseg.py",
-                    "ModelURL": source_dict["model_urls"],
-                }
-            ]
-        }
-        with open(splitext(fname_seg)[0] + ".json", "w") as fp:
-            json.dump(sidecar_json, fp, indent=4)
-
     if arguments.qc is not None:
         # If `arguments.qc_seg is None`, each entry will be treated as an
         # empty file with the same size as the corresponding input image
         qc_seg = [arguments.qc_seg] * len(input_filenames)
         # Models can have multiple input images -- create 1 QC report per input image.
         if len(output_filenames) == len(input_filenames):
             iterator = zip(input_filenames, output_filenames, [None] * len(input_filenames), qc_seg)
-        # Special case: totalspineseg which outputs 4-5 files per 1 input file
-        elif arguments.task == 'totalspineseg':
-            # `-step1-only: 1`: Use the 4th image ([3]) which represents the step1 output
-            if getattr(arguments, "step1_only") == 1:
-                assert len(output_filenames) == 4 * len(input_filenames)
-                output_filenames_qc = output_filenames[3::4]
-            # `-step1-only: 0`: Use the 5th image ([4]) which represents the step2 output
-            else:
-                assert len(output_filenames) == 5 * len(input_filenames)
-                output_filenames_qc = output_filenames[4::5]
-            iterator = zip(input_filenames, output_filenames_qc, [None] * len(input_filenames), qc_seg)
+        # Special case: spine which outputs 2-4 files per 1 input file
+        elif arguments.task == 'spine':
+            assert len(output_filenames) == 2 * len(input_filenames)
+            iterator = zip(input_filenames * 2, output_filenames, [None] * 2, qc_seg * 2)
```

Comment:
Right now, since we use `output_filenames` directly, this will produce 2 QC reports per input (one for the disc labels and one for the "full" output).

The "full" output looks normal, but the discs QC report is nearly impossible to see: